internal/backends/postgresql/metadata/registry.go

```diff
@@ -65,6 +81,32 @@ func NewRegistry(u string, l *zap.Logger, sp *state.Provider) (*Registry, error)
 		colls: map[string]map[string]*Collection{},
 	}
 
+	// use defer to close registry upon error to avoid forgetting
+	defer func() {
+		if err != nil {
+			r.Close()
```

Comment:
Minor comment: `r.p.Close()` might be a bit more safe here.
Currently, `r.Close()` only does `r.p.Close()` but if in the future we extend it and it will do something else that belongs to `r`, we might end up here by attempting to close something that is not initialized yet. But on the other hand, if we don't call `r.Close()` we also might end up by not closing something that must be closed. so I'm not sure :)

Reply:
I think implementation of `Close()` should handle closing something not initialised. Often `Close()` function sets variables to nil, but subsequent call to `Close()` shouldn't fail and should be idempotent. :hugs: 

```
func (r *Registry) Close() {
	r.p.Close()
	if r.something != nil {
		r.something.Close()
	}
}
```